Polyfills/URL/Source/URLSearchParams.cpp

```diff
@@ -0,0 +1,161 @@
+        }
+        std::string queryStr = info[0].As<Napi::String>();
+        parseInputQueryStr(queryStr);
+    }
+
+    void URLSearchParams::parseKeyVal(const std::string& subStr)
+    {
+        auto equalSign = subStr.find("=");
+        if (equalSign != std::string::npos)
+        {
+            auto key = subStr.substr(0, equalSign);
+            auto value = subStr.substr(equalSign + 1);
+
+            m_paramsVector.push_back(key);
+            m_paramsMap[key] = value;
+        }
+    }
+
+    void URLSearchParams::parseInputQueryStr(const std::string& queryStr)
+    {
+        // find all the &'s to determine where each param ends
+        // got positions for substring
+        if (queryStr.length() <= 1)
+        {
+            return;
+        }
+        
+        size_t start = queryStr[0] == '?' ? 1 : 0;
+        size_t end = queryStr.find("&");
+        // find the first &, in a while loop
+        while (end != std::string::npos)
+        {
+            // get substring
+            std::string mySub = queryStr.substr(start, end - start);
+            parseKeyVal(mySub);
+
+            // update start, end
+            start = end + 1;
+            end = queryStr.find("&", start);
+        }
+
+        // now need to parse the last key val pair
+        parseKeyVal(queryStr.substr(start));
+    }
+
+    Napi::Value URLSearchParams::Get(const Napi::CallbackInfo& info)
+    {
+        std::string key = info[0].As<Napi::String>();
+
+        auto element = m_paramsMap.find(key);
+
+        // element is not found 
+        if (element == m_paramsMap.end())
+        {
+            return Env().Null();
+        }
+        return Napi::Value::From(Env(), element->second);
+    }
+
+    std::string URLSearchParams::GetAllParams()
+    {
+        std::stringstream resultStringStream("");
+
+        if (m_paramsVector.empty())
+        {
+            return resultStringStream.str();
+        }
+
+        for (int i = 0; i < m_paramsVector.size(); i++)
+        {
+            if (i > 0) 
+            {
+                resultStringStream << "&";
+            }
+            else 
+            {
+                resultStringStream << "?";
+            }
+
+            std::string key = m_paramsVector[i];
+            std::string value = m_paramsMap[key];
+
+            resultStringStream << key;
+            resultStringStream << "=";
+            resultStringStream << value;
+        } 
+
+        return resultStringStream.str();
+    }
+
+
+    void URLSearchParams::Set(const Napi::CallbackInfo& info)
+    {
+        if (info.Length() < 2) 
+        {
+            return;
+        }
+
+        std::string key = info[0].As<Napi::String>();
+        std::string value = info[1].ToString().Utf8Value();
+
+        if (m_paramsMap.find(key) == m_paramsMap.end()) 
+        {
+            m_paramsVector.push_back(key);
+            m_paramsMap[key] = value;
+        }
+        else 
+        {
+            m_paramsMap[key] = value;
+        }
```

Comment:
Try using [insert_or_assign](https://en.cppreference.com/w/cpp/container/unordered_map/insert_or_assign) to eliminate two hash lookups.